### PR TITLE
Support call sounds on Safari

### PR DESCRIPTION
**Pull request by SystemKeeper**
### ☑️ Resolves

Until now we don't support call sounds on Safari. This has 2 reasons:
* Safari does not support .ogg
* Safari does only allow to play sounds if this is due to a user interaction

To solve the first issue we also provide FLAC files for the sounds which are working fine on Safari. To solve the second issue we create the audio object when we press the start call button. As this is done inside of a click event handler, the audio objects are free to play at any time. 
While this approach in theory also works iOS, it does not work anymore as soon as you grant camera and microphone permissions there - so we keep the note about it not being supported on those devices.

Follow-up to https://github.com/nextcloud/spreed/pull/5410
Tested on
* Safari 16.3
* Chrome 112
* Firefox 112

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: img/COPYING [hunk 1]
```diff
@@ -5,3 +5,11 @@
 ## leave_call.ogg
 @author Marco Ambrosini (marcoambrosini@icloud.com) 
 @license CC0-1.0
+
+## join_call.flac
```
Comment (nickvergessen) [inline img/COPYING:9]: you add 3 files but only 2 licenses?

File: src/store/soundsStore.js [hunk 2]
```diff
@@ -55,6 +60,76 @@ const mutations = {
 	setUserId(state, userId) {
 		state.userId = userId
 	},
+
+	/**
+	 * Sets the join audio object
+	 *
```
Comment (nickvergessen) [inline src/store/soundsStore.js:66]:
```suggestion
	 *
```

> ### GitHub Actions / eslint
>
> Trailing spaces not allowed                                              


Same for the other docs.

File: src/store/soundsStore.js [hunk 3]
```diff
@@ -55,6 +60,76 @@ const mutations = {
...
+		state.audioObjectsCreated = true
+	},
+}
+
+/**
+ * Helper method to get the full file path of an audio file in spreed/img.
+ * Example: getFullAudioFilepath('myAudio') => spreed/img/myAudio.ogg
+ * 
+ * @param {string} fileName The name of the file without extension
+ * @returns The full path to the file in the spreed/img directory and adds .ogg/.flac depending on supported codecs
```
Comment (nickvergessen) [inline src/store/soundsStore.js:109]:
```suggestion
 * @return The full path to the file in the spreed/img directory and adds .ogg/.flac depending on supported codecs
```

File: src/store/soundsStore.js [hunk 4]
```diff
@@ -55,6 +60,76 @@ const mutations = {
...
+ */
+const getFullAudioFilepath = function(fileName) {
+	const tempAudio = new Audio()
+
+	// Prefer the .ogg version of sounds, but fall back to .flac in case .ogg is not supported (Safari)
+	if (tempAudio.canPlayType("audio/ogg")) {
```
Comment (nickvergessen) [inline src/store/soundsStore.js:115]:
```suggestion
	if (tempAudio.canPlayType('audio/ogg')) {
```

File: src/store/soundsStore.js [hunk 5]
```diff
@@ -55,6 +60,76 @@ const mutations = {
...
+	} else {
+		return generateFilePath('spreed', 'img', fileName + '.flac')
+	}
+}
+
+/**
+ * Creates a HTMLAudioElement with the specified filePath and loads it
+ * 
+ * @param {string} filePath Path to the file
+ * @returns The created and loaded HTMLAudioElement
```
Comment (nickvergessen) [inline src/store/soundsStore.js:126]:
```suggestion
 * @return {HTMLAudioElement} The created and loaded HTMLAudioElement
```

File: img/COPYING (hunk 1 above)
Comment (SystemKeeper) [reply img/COPYING:9]: We did not specify a license for `LibremPhoneCall.ogg` before, so I did not add it for the FLAC version.

**Review by marcoambrosini — APPROVED** [review]
tested and works

**Review by marcoambrosini — APPROVED** [review]
tested and works

**Review by marcoambrosini — APPROVED** [review]
tested and works

File: img/COPYING (hunk 1 above)
Comment (SystemKeeper) [reply img/COPYING:9]: @marcoambrosini any idea about the third license ?

Comment (marcoambrosini) [reply img/COPYING:9]: No, that file was already there when I added the join and leave call sounds.

**Comment by SystemKeeper** [comment]
@nickvergessen Should this be backported?

**Comment by nickvergessen** [comment]
I think it would be quite a surprise for most users in a patch release. We can of course bump the minor version to indicate something a like, but otherwise I'm also fine with not backporting

**Pull request merged**